pkgs/sdk/server/src/Internal/DataSources/CompositeDataSource/CompositeSource.cs

```diff
@@ -0,0 +1,352 @@
+                // including calling back into the composite and queuing additional
+                // actions without blocking the queue.
+                // If an action throws an exception, catch it and continue processing
+                // the next action to prevent one failure from stopping all processing.
+                try
+                {
+                    action();
+                }
+                catch
+                {
+                    // Continue processing remaining actions even if one fails
+                    // TODO: need to add logging, will add in next PR
+                }
+            }
+        }
+
+        private void TryFindNextUnderLock()
+        {
+            // This method must only be called while holding _lock.
+            if (_currentDataSource != null)
+            {
+                return;
+            }
+
+            var entry = _sourcesList.Next();
+            if (entry.Factory == null)
+            {
+                return;
+            }
+
+            // Build the list of update sinks conditionally based on whether we have an action applier factory
+            var updateSinks = new List<IDataSourceUpdates>();
+            if (entry.ActionApplierFactory != null)
+            {
+                var actionApplier = entry.ActionApplierFactory.CreateActionApplier(this);
+                updateSinks.Add(actionApplier);
+            }
+            updateSinks.Add(_sanitizedUpdateSink);
+
+            // here we make a fanout so that we can trigger actions as well as forward calls to the sanitized sink (order matters here)
+            var fanout = new FanOutDataSourceUpdates(updateSinks);
+            var disableableUpdates = _disableableTracker.WrapAndTrack(fanout);
+            
+            _currentEntry = entry;
+            _currentDataSource = entry.Factory.CreateSource(disableableUpdates);
+        }
+
+        #region ICompositeSourceActionable
+
+        /// <summary>
+        /// Starts the current data source.
+        /// </summary>
+        public Task<bool> StartCurrent()
+        {
+            var tcs = new TaskCompletionSource<bool>();
+            
+            EnqueueAction(() =>
+            {
+                IDataSource dataSourceToStart;
+                lock (_lock)
+                {
+                    TryFindNextUnderLock();
+                    dataSourceToStart = _currentDataSource;
+                }
+
+                if (dataSourceToStart is null)
+                {
+                    // No sources available.
+                    tcs.SetResult(false);
+                    return;
+                }
+
+                // Start the source asynchronously and complete the task when it finishes.
+                // We do this outside the lock to avoid blocking the queue.
+                _ = Task.Run(async () =>
+                {
+                    try
+                    {
+                        var result = await dataSourceToStart.Start();
```

Comment:
```suggestion
                        var result = await dataSourceToStart.Start().ConfigureAwait(false);
```

The ConfigureAwait(false) means we don't care about the synchronization context we run on. It can be important if the code runs in something like if it was running in an HTTP context in ASP, or in an environment with winforms. The rule of thumb is basically to always do this, unless you know you need the same synchronization context, which is hardly ever the case.

Though in the client-side SDK it could be, if you were trying to do UI thread things being called from the UI thread. Which removes the need to like schedule something onto that thread, if you know the task will return on that thread.

I think this method specifically wouldn't matter, because you can never get to the entrypoint from caller code. Generally we should probably always use this setting. The reason I didn't want `ContinueWith` is because achieving the exact same thing is more complicated.